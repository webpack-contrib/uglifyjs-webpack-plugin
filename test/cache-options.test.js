import cacache from 'cacache';
import findCacheDir from 'find-cache-dir';
import UglifyJsPlugin from '../src/index';
import {
  PluginEnvironment,
  createCompiler,
  compile,
  cleanErrorStack,
} from './helpers';

const cacheDir = findCacheDir({ name: 'uglifyjs-webpack-plugin' });

describe('when options.cache', () => {
  const assets = {
    'test.js': {
      source: () => 'function test(foo) { foo = 1; }',
    },
    'test1.js': {
      source: () => 'function test1(foo) { foo = 1; }',
    },
    'test2.js': {
      source: () => 'function test2(foo) { foo = 1; }',
    },
    'test3.js': {
      source: () => 'function test3(foo) { foo = 1; }',
    },
  };

  describe('false', () => {
    let eventBindings;
    let eventBinding;

    beforeAll(() => cacache.rm.all(cacheDir));

    afterAll(() => cacache.rm.all(cacheDir));

    beforeEach(() => {
      const pluginEnvironment = new PluginEnvironment();
      const compilerEnv = pluginEnvironment.getEnvironmentStub();
      compilerEnv.context = '';

      const plugin = new UglifyJsPlugin({
        cache: false,
      });
      plugin.apply(compilerEnv);
      eventBindings = pluginEnvironment.getEventBindings();
    });

    it('binds one event handler', () => {
      expect(eventBindings.length).toBe(1);
    });

    describe('compilation handler', () => {
      beforeEach(() => {
        [eventBinding] = eventBindings;
      });

      it('binds to compilation event', () => {
        expect(eventBinding.name).toBe('compilation');
      });

      describe('when called', () => {
        let chunkPluginEnvironment;
        let compilationEventBindings;
        let compilationEventBinding;
        let compilation;
        let callback;

        beforeEach(() => {
          chunkPluginEnvironment = new PluginEnvironment();
          compilation = chunkPluginEnvironment.getEnvironmentStub();
          compilation.assets = Object.assign({}, assets);
          compilation.errors = [];

          eventBinding.handler(compilation);
          compilationEventBindings = chunkPluginEnvironment.getEventBindings();
        });

        it('binds one event handler', () => {
          expect(compilationEventBindings.length).toBe(1);
        });

        describe('optimize-chunk-assets handler', () => {
          beforeEach(() => {
            [compilationEventBinding] = compilationEventBindings;
          });

          it('binds to optimize-chunk-assets event', () => {
            expect(compilationEventBinding.name).toEqual('optimize-chunk-assets');
          });

          it('only calls callback once', (done) => {
            callback = jest.fn();
            compilationEventBinding.handler([''], () => {
              callback();
              expect(callback.mock.calls.length).toBe(1);
              done();
            });
          });

          it('cache files', (done) => {
            const files = ['test.js', 'test1.js', 'test2.js', 'test3.js'];

            cacache.get = jest.fn(cacache.get);
            cacache.put = jest.fn(cacache.put);

            compilationEventBinding.handler([{
              files,
            }], () => {
              // Cache disabled so we don't run `get` or `put`
              expect(cacache.get.mock.calls.length).toBe(0);
              expect(cacache.put.mock.calls.length).toBe(0);

              cacache
                .ls(cacheDir)
                .then((cacheEntriesList) => {
                  const cacheKeys = Object.keys(cacheEntriesList);

                  expect(cacheKeys.length).toBe(0);
                  done();
                });
            });
          });
        });
      });
    });

    it('matches snapshot', () => {
      const compiler = createCompiler();
      new UglifyJsPlugin({ cache: true }).apply(compiler);

      return compile(compiler)
        .then((stats) => {
          const errors = stats.compilation.errors.map(cleanErrorStack);
          const warnings = stats.compilation.warnings.map(cleanErrorStack);

          expect(errors).toMatchSnapshot('errors');
          expect(warnings).toMatchSnapshot('warnings');

          for (const file in stats.compilation.assets) {
            if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
              expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
            }
          }
        });
    });
  });

  describe('true', () => {
    let eventBindings;
    let eventBinding;

    beforeAll(() => cacache.rm.all(cacheDir));

    afterAll(() => cacache.rm.all(cacheDir));

    beforeEach(() => {
      const pluginEnvironment = new PluginEnvironment();
      const compilerEnv = pluginEnvironment.getEnvironmentStub();
      compilerEnv.context = '';

      const plugin = new UglifyJsPlugin({
        cache: true,
      });
      plugin.apply(compilerEnv);
      eventBindings = pluginEnvironment.getEventBindings();
    });

    it('binds one event handler', () => {
      expect(eventBindings.length).toBe(1);
    });

    describe('compilation handler', () => {
      beforeEach(() => {
        [eventBinding] = eventBindings;
      });

      it('binds to compilation event', () => {
        expect(eventBinding.name).toBe('compilation');
      });

      describe('when called', () => {
        let chunkPluginEnvironment;
        let compilationEventBindings;
        let compilationEventBinding;
        let compilation;
        let callback;

        beforeEach(() => {
          chunkPluginEnvironment = new PluginEnvironment();
          compilation = chunkPluginEnvironment.getEnvironmentStub();
          compilation.assets = Object.assign({}, assets);
          compilation.errors = [];

          eventBinding.handler(compilation);
          compilationEventBindings = chunkPluginEnvironment.getEventBindings();
        });

        it('binds one event handler', () => {
          expect(compilationEventBindings.length).toBe(1);
        });

        describe('optimize-chunk-assets handler', () => {
          beforeEach(() => {
            [compilationEventBinding] = compilationEventBindings;
          });

          it('binds to optimize-chunk-assets event', () => {
            expect(compilationEventBinding.name).toEqual('optimize-chunk-assets');
          });

          it('only calls callback once', (done) => {
            callback = jest.fn();
            compilationEventBinding.handler([''], () => {
              callback();
              expect(callback.mock.calls.length).toBe(1);
              done();
            });
          });

          it('cache files', (done) => {
            const files = ['test.js', 'test1.js', 'test2.js', 'test3.js'];

            cacache.get = jest.fn(cacache.get);
            cacache.put = jest.fn(cacache.put);

            compilationEventBinding.handler([{
              files,
            }], () => {
              // Try to found cached files, but we don't have their in cache
              expect(cacache.get.mock.calls.length).toBe(4);
              // Put files in cache
              expect(cacache.put.mock.calls.length).toBe(4);

              cacache
                .ls(cacheDir)
                .then((cacheEntriesList) => {
                  const cacheKeys = Object.keys(cacheEntriesList);

                  // Make sure that we cached files
                  expect(cacheKeys.length).toBe(files.length);
                  cacheKeys.forEach((cacheEntry) => {
                    // eslint-disable-next-line no-new-func
                    const cacheEntryOptions = new Function(`'use strict'\nreturn ${cacheEntry}`)();

                    expect([cacheEntryOptions.path, cacheEntryOptions.hash])
                      .toMatchSnapshot(cacheEntryOptions.path);
                  });

                  // Reset compilation assets and mocks
                  compilation.assets = Object.assign({}, assets);
                  compilation.errors = [];

                  cacache.get.mockClear();
                  cacache.put.mockClear();

                  compilationEventBinding.handler([{
                    files,
                  }], () => {
                    // Now we have cached files so we get their and don't put
                    expect(cacache.get.mock.calls.length).toBe(4);
                    expect(cacache.put.mock.calls.length).toBe(0);

                    done();
                  });
                });
            });
          });
        });
      });
    });

    it('matches snapshot', () => {
      const compiler = createCompiler();
      new UglifyJsPlugin({ cache: true }).apply(compiler);

      return compile(compiler)
        .then((stats) => {
          const errors = stats.compilation.errors.map(cleanErrorStack);
          const warnings = stats.compilation.warnings.map(cleanErrorStack);

          expect(errors).toMatchSnapshot('errors');
          expect(warnings).toMatchSnapshot('warnings');

          for (const file in stats.compilation.assets) {
            if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
              expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
            }
          }
        });
    });
  });

  describe('string', () => {
    const othercacheDir = findCacheDir({ name: 'other-cache-directory' });
    let eventBindings;
    let eventBinding;

    beforeAll(() => cacache.rm.all(othercacheDir));

    afterAll(() => cacache.rm.all(othercacheDir));

    beforeEach(() => {
      const pluginEnvironment = new PluginEnvironment();
      const compilerEnv = pluginEnvironment.getEnvironmentStub();
      compilerEnv.context = '';

      const plugin = new UglifyJsPlugin({
        cache: othercacheDir,
      });
      plugin.apply(compilerEnv);
      eventBindings = pluginEnvironment.getEventBindings();
    });

    it('binds one event handler', () => {
      expect(eventBindings.length).toBe(1);
    });

    describe('compilation handler', () => {
      beforeEach(() => {
        [eventBinding] = eventBindings;
      });

      it('binds to compilation event', () => {
        expect(eventBinding.name).toBe('compilation');
      });

      describe('when called', () => {
        let chunkPluginEnvironment;
        let compilationEventBindings;
        let compilationEventBinding;
        let compilation;
        let callback;

        beforeEach(() => {
          chunkPluginEnvironment = new PluginEnvironment();
          compilation = chunkPluginEnvironment.getEnvironmentStub();
          compilation.assets = Object.assign({}, assets);
          compilation.errors = [];

          eventBinding.handler(compilation);
          compilationEventBindings = chunkPluginEnvironment.getEventBindings();
        });

        it('binds one event handler', () => {
          expect(compilationEventBindings.length).toBe(1);
        });

        describe('optimize-chunk-assets handler', () => {
          beforeEach(() => {
            [compilationEventBinding] = compilationEventBindings;
          });

          it('binds to optimize-chunk-assets event', () => {
            expect(compilationEventBinding.name).toEqual('optimize-chunk-assets');
          });

          it('only calls callback once', (done) => {
            callback = jest.fn();
            compilationEventBinding.handler([''], () => {
              callback();
              expect(callback.mock.calls.length).toBe(1);
              done();
            });
          });

          it('cache files', (done) => {
            const files = ['test.js', 'test1.js', 'test2.js', 'test3.js'];

            cacache.get = jest.fn(cacache.get);
            cacache.put = jest.fn(cacache.put);

            compilationEventBinding.handler([{
              files,
            }], () => {
              // Try to found cached files, but we don't have their in cache
              expect(cacache.get.mock.calls.length).toBe(4);
              // Put files in cache
              expect(cacache.put.mock.calls.length).toBe(4);

              cacache
                .ls(othercacheDir)
                .then((cacheEntriesList) => {
                  const cacheKeys = Object.keys(cacheEntriesList);

                  // Make sure that we cached files
                  expect(cacheKeys.length).toBe(files.length);
                  cacheKeys.forEach((cacheEntry) => {
                    // eslint-disable-next-line no-new-func
                    const cacheEntryOptions = new Function(`'use strict'\nreturn ${cacheEntry}`)();

                    expect([cacheEntryOptions.path, cacheEntryOptions.hash])
                      .toMatchSnapshot(cacheEntryOptions.path);
                  });

                  // Reset compilation assets and mocks
                  compilation.assets = Object.assign({}, assets);
                  compilation.errors = [];

                  cacache.get.mockClear();
                  cacache.put.mockClear();

                  compilationEventBinding.handler([{
                    files,
                  }], () => {
                    // Now we have cached files so we get their and don't put
                    expect(cacache.get.mock.calls.length).toBe(4);
                    expect(cacache.put.mock.calls.length).toBe(0);

                    done();
                  });
                });
            });
          });
        });
      });
    });

    it('matches snapshot', () => {
      const compiler = createCompiler();
      new UglifyJsPlugin({ cache: othercacheDir }).apply(compiler);

      return compile(compiler)
        .then((stats) => {
          const errors = stats.compilation.errors.map(cleanErrorStack);
          const warnings = stats.compilation.warnings.map(cleanErrorStack);

          expect(errors).toMatchSnapshot('errors');
          expect(warnings).toMatchSnapshot('warnings');

          for (const file in stats.compilation.assets) {
            if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
              expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
            }
          }
        });
    });
  });

  describe('when options.additionalCacheKeys', () => {
    let eventBindings;
    let eventBinding;

    beforeAll(() => cacache.rm.all(cacheDir));

    afterAll(() => cacache.rm.all(cacheDir));

    beforeEach(() => {
      const pluginEnvironment = new PluginEnvironment();
      const compilerEnv = pluginEnvironment.getEnvironmentStub();
      compilerEnv.context = '';

      const plugin = new UglifyJsPlugin({
        cache: true,
        additionalCacheKeys: {
          myCustomKey: 'myCustomValue',
        },
      });
      plugin.apply(compilerEnv);
      eventBindings = pluginEnvironment.getEventBindings();
    });

    it('binds one event handler', () => {
      expect(eventBindings.length).toBe(1);
    });

    describe('compilation handler', () => {
      beforeEach(() => {
        [eventBinding] = eventBindings;
      });

      it('binds to compilation event', () => {
        expect(eventBinding.name).toBe('compilation');
      });

      describe('when called', () => {
        let chunkPluginEnvironment;
        let compilationEventBindings;
        let compilationEventBinding;
        let compilation;
        let callback;

        beforeEach(() => {
          chunkPluginEnvironment = new PluginEnvironment();
          compilation = chunkPluginEnvironment.getEnvironmentStub();
          compilation.assets = Object.assign({}, assets);
          compilation.errors = [];

          eventBinding.handler(compilation);
          compilationEventBindings = chunkPluginEnvironment.getEventBindings();
        });

        it('binds one event handler', () => {
          expect(compilationEventBindings.length).toBe(1);
        });

        describe('optimize-chunk-assets handler', () => {
          beforeEach(() => {
            [compilationEventBinding] = compilationEventBindings;
          });

          it('binds to optimize-chunk-assets event', () => {
            expect(compilationEventBinding.name).toEqual('optimize-chunk-assets');
          });

          it('only calls callback once', (done) => {
            callback = jest.fn();
            compilationEventBinding.handler([''], () => {
              callback();
              expect(callback.mock.calls.length).toBe(1);
              done();
            });
          });

          it('cache files', (done) => {
            const files = ['test.js', 'test1.js', 'test2.js', 'test3.js'];

            cacache.get = jest.fn(cacache.get);
            cacache.put = jest.fn(cacache.put);

            compilationEventBinding.handler([{
              files,
            }], () => {
              // Try to found cached files, but we don't have their in cache
              expect(cacache.get.mock.calls.length).toBe(4);
              // Put files in cache
              expect(cacache.put.mock.calls.length).toBe(4);

              cacache
                .ls(cacheDir)
                .then((cacheEntriesList) => {
                  const cacheKeys = Object.keys(cacheEntriesList);

                  // Make sure that we cached files
                  expect(cacheKeys.length).toBe(files.length);
                  cacheKeys.forEach((cacheEntry) => {
                    // eslint-disable-next-line no-new-func
                    const cacheEntryOptions = new Function(`'use strict'\nreturn ${cacheEntry}`)();

                    expect([cacheEntryOptions.path, cacheEntryOptions.hash])
                      .toMatchSnapshot(cacheEntryOptions.path);
                  });

                  // Reset compilation assets and mocks
                  compilation.assets = Object.assign({}, assets);
                  compilation.errors = [];

                  cacache.get.mockClear();
                  cacache.put.mockClear();

                  compilationEventBinding.handler([{
                    files,
                  }], () => {
                    // Now we have cached files so we get their and don't put
                    expect(cacache.get.mock.calls.length).toBe(4);
                    expect(cacache.put.mock.calls.length).toBe(0);

                    done();
                  });
                });
            });
          });
        });
      });
    });

    it('matches snapshot', () => {
      const compiler = createCompiler();
      new UglifyJsPlugin({ cache: true }).apply(compiler);

      return compile(compiler)
        .then((stats) => {
          const errors = stats.compilation.errors.map(cleanErrorStack);
          const warnings = stats.compilation.warnings.map(cleanErrorStack);

          expect(errors).toMatchSnapshot('errors');
          expect(warnings).toMatchSnapshot('warnings');

          for (const file in stats.compilation.assets) {
            if (Object.prototype.hasOwnProperty.call(stats.compilation.assets, file)) {
              expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
            }
          }
        });
    });
  });
});
