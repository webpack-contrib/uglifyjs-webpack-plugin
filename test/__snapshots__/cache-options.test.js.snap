// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when options.cache false matches snapshot: errors 1`] = `Array []`;

exports[`when options.cache false matches snapshot: main.0c220ec66316af2c1b24.js 1`] = `"webpackJsonp([0],[function(o,n){o.exports=function(){console.log(7)}}],[0]);"`;

exports[`when options.cache false matches snapshot: manifest.d6857f782c13a99b5917.js 1`] = `"!function(r){var n=window.webpackJsonp;window.webpackJsonp=function(e,u,c){for(var f,i,p,a=0,l=[];a<e.length;a++)i=e[a],o[i]&&l.push(o[i][0]),o[i]=0;for(f in u)Object.prototype.hasOwnProperty.call(u,f)&&(r[f]=u[f]);for(n&&n(e,u,c);l.length;)l.shift()();if(c)for(a=0;a<c.length;a++)p=t(t.s=c[a]);return p};var e={},o={1:0};function t(n){if(e[n])return e[n].exports;var o=e[n]={i:n,l:!1,exports:{}};return r[n].call(o.exports,o,o.exports,t),o.l=!0,o.exports}t.m=r,t.c=e,t.d=function(r,n,e){t.o(r,n)||Object.defineProperty(r,n,{configurable:!1,enumerable:!0,get:e})},t.n=function(r){var n=r&&r.__esModule?function(){return r.default}:function(){return r};return t.d(n,\\"a\\",n),n},t.o=function(r,n){return Object.prototype.hasOwnProperty.call(r,n)},t.p=\\"\\",t.oe=function(r){throw console.error(r),r}}([]);"`;

exports[`when options.cache false matches snapshot: warnings 1`] = `Array []`;

exports[`when options.cache string compilation handler when called optimize-chunk-assets handler cache files: test.js 1`] = `
Array [
  "test.js",
  "28e4d31993a4f0bf5420d06d6e6013b8",
]
`;

exports[`when options.cache string compilation handler when called optimize-chunk-assets handler cache files: test1.js 1`] = `
Array [
  "test1.js",
  "0ee8d6a0ecdd7bc2322d1e726ef41cdd",
]
`;

exports[`when options.cache string compilation handler when called optimize-chunk-assets handler cache files: test2.js 1`] = `
Array [
  "test2.js",
  "ba96de0d701dab1af6ca647f97d42fb7",
]
`;

exports[`when options.cache string compilation handler when called optimize-chunk-assets handler cache files: test3.js 1`] = `
Array [
  "test3.js",
  "ef8b436e106d1bbfe07b50263e91deac",
]
`;

exports[`when options.cache string matches snapshot: errors 1`] = `Array []`;

exports[`when options.cache string matches snapshot: main.0c220ec66316af2c1b24.js 1`] = `"webpackJsonp([0],[function(o,n){o.exports=function(){console.log(7)}}],[0]);"`;

exports[`when options.cache string matches snapshot: manifest.d6857f782c13a99b5917.js 1`] = `"!function(r){var n=window.webpackJsonp;window.webpackJsonp=function(e,u,c){for(var f,i,p,a=0,l=[];a<e.length;a++)i=e[a],o[i]&&l.push(o[i][0]),o[i]=0;for(f in u)Object.prototype.hasOwnProperty.call(u,f)&&(r[f]=u[f]);for(n&&n(e,u,c);l.length;)l.shift()();if(c)for(a=0;a<c.length;a++)p=t(t.s=c[a]);return p};var e={},o={1:0};function t(n){if(e[n])return e[n].exports;var o=e[n]={i:n,l:!1,exports:{}};return r[n].call(o.exports,o,o.exports,t),o.l=!0,o.exports}t.m=r,t.c=e,t.d=function(r,n,e){t.o(r,n)||Object.defineProperty(r,n,{configurable:!1,enumerable:!0,get:e})},t.n=function(r){var n=r&&r.__esModule?function(){return r.default}:function(){return r};return t.d(n,\\"a\\",n),n},t.o=function(r,n){return Object.prototype.hasOwnProperty.call(r,n)},t.p=\\"\\",t.oe=function(r){throw console.error(r),r}}([]);"`;

exports[`when options.cache string matches snapshot: warnings 1`] = `Array []`;

exports[`when options.cache true compilation handler when called optimize-chunk-assets handler cache files: test.js 1`] = `
Array [
  "test.js",
  "28e4d31993a4f0bf5420d06d6e6013b8",
]
`;

exports[`when options.cache true compilation handler when called optimize-chunk-assets handler cache files: test1.js 1`] = `
Array [
  "test1.js",
  "0ee8d6a0ecdd7bc2322d1e726ef41cdd",
]
`;

exports[`when options.cache true compilation handler when called optimize-chunk-assets handler cache files: test2.js 1`] = `
Array [
  "test2.js",
  "ba96de0d701dab1af6ca647f97d42fb7",
]
`;

exports[`when options.cache true compilation handler when called optimize-chunk-assets handler cache files: test3.js 1`] = `
Array [
  "test3.js",
  "ef8b436e106d1bbfe07b50263e91deac",
]
`;

exports[`when options.cache true matches snapshot: errors 1`] = `Array []`;

exports[`when options.cache true matches snapshot: main.0c220ec66316af2c1b24.js 1`] = `"webpackJsonp([0],[function(o,n){o.exports=function(){console.log(7)}}],[0]);"`;

exports[`when options.cache true matches snapshot: manifest.d6857f782c13a99b5917.js 1`] = `"!function(r){var n=window.webpackJsonp;window.webpackJsonp=function(e,u,c){for(var f,i,p,a=0,l=[];a<e.length;a++)i=e[a],o[i]&&l.push(o[i][0]),o[i]=0;for(f in u)Object.prototype.hasOwnProperty.call(u,f)&&(r[f]=u[f]);for(n&&n(e,u,c);l.length;)l.shift()();if(c)for(a=0;a<c.length;a++)p=t(t.s=c[a]);return p};var e={},o={1:0};function t(n){if(e[n])return e[n].exports;var o=e[n]={i:n,l:!1,exports:{}};return r[n].call(o.exports,o,o.exports,t),o.l=!0,o.exports}t.m=r,t.c=e,t.d=function(r,n,e){t.o(r,n)||Object.defineProperty(r,n,{configurable:!1,enumerable:!0,get:e})},t.n=function(r){var n=r&&r.__esModule?function(){return r.default}:function(){return r};return t.d(n,\\"a\\",n),n},t.o=function(r,n){return Object.prototype.hasOwnProperty.call(r,n)},t.p=\\"\\",t.oe=function(r){throw console.error(r),r}}([]);"`;

exports[`when options.cache true matches snapshot: warnings 1`] = `Array []`;

exports[`when options.cache when options.additionalCacheKeys compilation handler when called optimize-chunk-assets handler cache files: test.js 1`] = `
Array [
  "test.js",
  "28e4d31993a4f0bf5420d06d6e6013b8",
]
`;

exports[`when options.cache when options.additionalCacheKeys compilation handler when called optimize-chunk-assets handler cache files: test1.js 1`] = `
Array [
  "test1.js",
  "0ee8d6a0ecdd7bc2322d1e726ef41cdd",
]
`;

exports[`when options.cache when options.additionalCacheKeys compilation handler when called optimize-chunk-assets handler cache files: test2.js 1`] = `
Array [
  "test2.js",
  "ba96de0d701dab1af6ca647f97d42fb7",
]
`;

exports[`when options.cache when options.additionalCacheKeys compilation handler when called optimize-chunk-assets handler cache files: test3.js 1`] = `
Array [
  "test3.js",
  "ef8b436e106d1bbfe07b50263e91deac",
]
`;

exports[`when options.cache when options.additionalCacheKeys matches snapshot: errors 1`] = `Array []`;

exports[`when options.cache when options.additionalCacheKeys matches snapshot: main.0c220ec66316af2c1b24.js 1`] = `"webpackJsonp([0],[function(o,n){o.exports=function(){console.log(7)}}],[0]);"`;

exports[`when options.cache when options.additionalCacheKeys matches snapshot: manifest.d6857f782c13a99b5917.js 1`] = `"!function(r){var n=window.webpackJsonp;window.webpackJsonp=function(e,u,c){for(var f,i,p,a=0,l=[];a<e.length;a++)i=e[a],o[i]&&l.push(o[i][0]),o[i]=0;for(f in u)Object.prototype.hasOwnProperty.call(u,f)&&(r[f]=u[f]);for(n&&n(e,u,c);l.length;)l.shift()();if(c)for(a=0;a<c.length;a++)p=t(t.s=c[a]);return p};var e={},o={1:0};function t(n){if(e[n])return e[n].exports;var o=e[n]={i:n,l:!1,exports:{}};return r[n].call(o.exports,o,o.exports,t),o.l=!0,o.exports}t.m=r,t.c=e,t.d=function(r,n,e){t.o(r,n)||Object.defineProperty(r,n,{configurable:!1,enumerable:!0,get:e})},t.n=function(r){var n=r&&r.__esModule?function(){return r.default}:function(){return r};return t.d(n,\\"a\\",n),n},t.o=function(r,n){return Object.prototype.hasOwnProperty.call(r,n)},t.p=\\"\\",t.oe=function(r){throw console.error(r),r}}([]);"`;

exports[`when options.cache when options.additionalCacheKeys matches snapshot: warnings 1`] = `Array []`;
